lib/src/facts/openbsd/networking_resolver.cc

```diff
@@ -0,0 +1,185 @@
+#include <internal/facts/openbsd/networking_resolver.hpp>
+#include <internal/util/bsd/scoped_ifaddrs.hpp>
+#include <facter/execution/execution.hpp>
+#include <facter/util/file.hpp>
+#include <facter/util/directory.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+#include <sys/sockio.h>
+#include <sys/ioctl.h>
+#include <net/if.h>
+#include <net/if_dl.h>
+#include <netinet/in.h>
+
+using namespace std;
+using namespace facter::util;
+using namespace facter::util::bsd;
+using namespace facter::execution;
+
+namespace facter { namespace facts { namespace openbsd {
+
+    networking_resolver::data networking_resolver::collect_data(collection& facts)
```

Comment:
I think that would be acceptable, yes.

It seems that `populate_address()` is a private function so now trying to figure out how to clean this up and get it working again.